packages/hyperion-autologging/src/ALSurfaceProxy.ts

```diff
@@ -35,21 +34,23 @@ type ProxyInitOptions =
  * If we can find a way around this limitation, we can use a simpler logic
  * like the following:
  */
-function SurfaceProxy(props: React.PropsWithChildren<ProxyInitOptions>) {
+function SurfaceProxy(props: React.PropsWithChildren<ProxyInitOptions>): React.ReactElement {
```

Comment:
you can return `React.Node` here, which is the same type as `props.children`.

Reply:
Typescript was complaining even before the return type was added.    I tried returning just `children` but children can be undefined (even though we check node != null),  which causes a type error in the interception `createElement`.

```
type React.ReactNode = string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal | null | undefined
```
<img width="631" alt="image" src="https://user-images.githubusercontent.com/6064408/221892879-190b86f5-e0b1-4a37-b912-215fb73a1d96.png">

```
Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.ts(2769)
```

Reply:
What I just pushed won't work either,  getting `throwInvalidHookError`.